### PR TITLE
Milestones 3 : Query multiple tables

### PR DESCRIPTION
**Pull request by BenMKT**
## Project scope:

- For this milestone of my project, I added some new tables and foreign key columns to the existing animal's table to model one-to-many relationships. 
- Used JOIN to query the data.

Below attached are the respective Screenshots of my work:

Animals owned by Dean Winchester not escaped
Animals per species table1
Create owners table1
Create species table1
Insert data to owners table1
Insert data to species table1
Insert owner_id value to animals table1
Insert species_id value to animals table1
List all Digimon owned by Jennifer Orwell1
List all owners and their animals1
List Animals of type Pokeman1
Modify animals table1
Owns most animals

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -48,3 +48,48 @@ BEGIN TRANSACTION;
 UPDATE animals SET weight_kg = 20.4 WHERE name = 'Boarmon';
 SELECT * FROM animals;
 COMMIT;
+
+ALTER TABLE animals DROP COLUMN species;
+ALTER TABLE animals ADD COLUMN species_id INT;
+ALTER TABLE animals ADD COLUMN owner_id INT;
+ALTER TABLE animals ADD CONSTRAINT fk_species FOREIGN KEY (species_id) REFERENCES species(id);
```
Comment (Shedrack-Sunday) [inline queries.sql]: - [ ] These changes should be done in `schema.sql` as suggested in your project requirements. This is the best practice as your table creation is there 👍 

### View:
image

File: data.sql [hunk 2]
```diff
@@ -15,4 +15,20 @@ VALUES
...
+VALUES
+('Sam Smith', 34),
+('Jennifer Orwell', 19),
+('Bob', 45),
+('Melody Pond', 77),
+('Dean Winchester', 14),
+('Jodie Whittaker', 38);
+
+--INSERT DATA INTO species TABLE
+INSERT INTO species (name)
```
Comment (Shedrack-Sunday) [inline data.sql:31]: - [ ] You missed a very key requirement that should be done on the `owner`. Kindl find them below, and please add the screenshot to the pull request

### Missed requirement:
image

File: queries.sql [hunk 3]
```diff
@@ -48,3 +48,48 @@ BEGIN TRANSACTION;
...
+  SELECT o.id
+  FROM owners o
+  WHERE o.full_name = CASE
+    WHEN animals.name = 'Agumon' THEN 'Sam Smith'
+    WHEN animals.name IN ('Gabumon', 'Pikachu') THEN 'Jennifer Orwell'
+    WHEN animals.name IN ('Devimon', 'Plantmon') THEN 'Bob'
+    WHEN animals.name IN ('Charmander', 'Squirtle', 'Blossom') THEN 'Melody Pond'
+    WHEN animals.name IN ('Angemon', 'Boarmon') THEN 'Dean Winchester'
+    ELSE NULL
+  END
```
Comment (Shedrack-Sunday) [inline queries.sql]: - [x] Kindly make these changes in the `data.sql file. I noted it's missing there, but kindly transfer these changes there. Its best practice for separating all three files with their functions 👍

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @BenMKT         👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the SQL features ✔️
- Good Use of testing query parameters ✔️ 
    
   

# Required Changes  🌵  

- [ ] You've done well, However, kindly add the screenshots of missing queries i mentioned in the comments below, and also transfer the SQL to the appropriate files 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 1 above)
Comment (BenMKT) [reply queries.sql]: - Moved statements from the `queries.sql` file into `schema.sql` file with the requested modifications to resolve. 
- Screenshots attached:
Code Review Alter Tables
Code Review ID Defaults updates

File: data.sql (hunk 2 above)
Comment (BenMKT) [reply data.sql:31]: - Moved files from `queries.sql` file to `data.sql` file to resolve above.

File: queries.sql (hunk 3 above)
Comment (BenMKT) [reply queries.sql]: -  Moved files from `queries.sql` file to `data.sql` file to resolve bug.

File: queries.sql (hunk 1 above)
Comment (peterrobert) [reply queries.sql]: This change has been done well

File: queries.sql (hunk 3 above)
Comment (peterrobert) [reply queries.sql]: This change has been done well

File: data.sql (hunk 2 above)
Comment (peterrobert) [reply data.sql:31]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @BenMKT , 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work making the changes requested by the previous reviewer ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**